### ISSUE TITLE
"Locking" Packages?

### ISSUE THREAD

**Issue by Eldog02** [quat1024/packages#13]
I love Packages, they look really nice compared to similar mods and the restrictions in play make them much more interesting to use in my opinion-- I really like using them in tandem with Create, because they make great buffers for items.

Would you consider some sort of "locking" mechanism? The idea is that packages could keep 0 of a specific item in them, and reject any other items attempted to be inserted. This could let packages act as basic filters.

As a basic visual indicator, the "0" on the front of a package could turn red, like when one is full. Or you could physically slap a little lock onto it, idk.

**Comment by quat1024** [comment]
This is something I'm not sure about. I intentionally left any form of locking out of the mod when initially designing it, but over the past couple days i've been chewing on it and trying to figure out where packages wants to be on the "convenient for everyday use" <-> "redstone assholery funtimes" spectrum lol

one of the Things™️ is that packages are inherently kinda stackable and nestable; if "locked status" was a property of the package it would have to go on the NBT tag, which would make them different items. It'd have to be a transient in-world thing

But theres also the "If you don't take the last couple items out it's basically the same as locking 😉 just do some redstone ✨" argument lol

**Comment by Eldog02** [comment]
Yeah, the "just keep the last item in lol" argument is fairly strong, I think it just depends on how much you value ease of use over the "complexity is fun" stance.

Not that my individual opinion rlly matters, but personally, I feel like this is a small thing that is usually just a bit obnoxious to work around. The redstone for it is typically fairly straightforward, it just takes up a bunch of space and isn't that interesting a challenge.

Ultimately I didn't really have any expectation as to whether this would be implemented or not, since I was aware that the mod is restrictive by design. 'S just my two cents.

**Comment by quat1024** [comment]
Hi it's 3 years later, I've been dusting this mod off in the [new repo](https://github.com/quat1024/mods) and I've decided to be less of an ass about package locking. In that commit ^ I added the ability for packages to be sticky if they're placed next to a slime block. Always thought that was a bit silly. So now it'll just be an item.

<img width="1960" height="534" alt="Image" src="https://github.com/user-attachments/assets/139d56a1-df18-47d2-bc3b-82abfff43097" />

The stickiness state isn't saved to item NBT at all. So there won't be any problems with package stacking in the inventory. Pretty simple.

 will release pretty soon for 1.20.1 (and maybe later?) if all goes well.

**Comment by quat1024** [comment]
Implemented in packages v2025.08.21 (part of Season 2)
